File: api-factory/af-encryption-gateway/src/main/java/com/example/afencryptiongateway/exceptionhandler/GlobalExceptionHandler.java
package com.example.afencryptiongateway.exceptionhandler;

import com.example.afencryptiongateway.dto.response.R;
import com.example.afencryptiongateway.exception.ForbidRequestException;
import jakarta.validation.constraints.NotNull;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice
@Slf4j
public class GlobalExceptionHandler {

    @ResponseBody
    @ResponseStatus(code = HttpStatus.FORBIDDEN)
    @ExceptionHandler({ForbidRequestException.class})
    public R<Object> handleForbidRequestException(@NotNull ForbidRequestException e) {
        return R.forbidden(e.getMessage());
    }
}

File: api-factory/af-security-gateway/src/main/kotlin/org/inet/aet/afsecuritygateway/exception/handler/GlobalExceptionHandler.kt
package org.inet.aet.afsecuritygateway.exception.handler

import org.inet.aet.afsecuritygateway.dto.response.R
import org.inet.aet.afsecuritygateway.dto.response.R_FORBIIDEN
import org.inet.aet.afsecuritygateway.exception.AbsentPermissionException
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.ResponseBody
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class GlobalExceptionHandler {

    @ResponseBody
    @ResponseStatus(code = HttpStatus.FORBIDDEN)
    @ExceptionHandler(AbsentPermissionException::class)
    fun handleAbsentPermissionException(e: AbsentPermissionException): R<Any> {
        return R_FORBIIDEN("权限不足，禁止访问")
    }
}